### PR TITLE
[dev] Upgrade GnuPG component packages to latest versions

### PR DESCRIPTION
**Pull request by thcrain-msft**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 
Upgrades the GnuPG component packages to their latest versions.

###### Change Log  
- All spec were linted, all licenses were verified for new versions, and build/runtime requirements were more fully specified wherever possible.
- gnupg2: upgrade from 2.2.20 to 2.3.3
- gpgme: upgrade from 1.13.1 to 1.16.0
- libassuan: upgrade from 2.5.1 to 2.5.5, add devel package in style of other system library packages
- libgcrypt: upgrade from 1.8.7 to 1.9.4
- libksba: upgrade from 1.3.5 to 1.6.0, add devel package in style of other system library packages
- pinentry: upgrade from 1.1.0 to 1.2.0
- npth: Change source URL from GitHub to official GnuPG server

###### Does this affect the toolchain?  

**YES**

###### Test Methodology

- Local toolchain build

### CODE REVIEW

File: SPECS/gpgme/gpgme.spec [hunk 1]
```diff
@@ -54,76 +52,99 @@ sed -i 's/defined(__FreeBSD__)/defined(__FreeBSD__) || defined(__GLIBC__)/g' src
...
 cd tests
-make check-TESTS
+%make_build check-TESTS
 
 %files
 %defattr(-,root,root)
-%license COPYING
-%{_libdir}/*.so.*
+%license AUTHORS COPYING COPYING.LESSER LICENSES
+%{_libdir}/*.so.11*
```
Comment (PawelWMS) [inline SPECS/gpgme/gpgme.spec:67]: Nice.:)

File: SPECS/libassuan/libassuan.spec [hunk 2]
```diff
@@ -21,61 +19,80 @@ libassuan is not, however, limited to use with GnuPG servers and clients.
...
+%{_libdir}/%{name}.so
+%{_libdir}/pkgconfig/%{name}.pc
+%{_includedir}/assuan.h
 %{_datadir}/aclocal/*
 
 %changelog
-* Mon Sep 28 2020 Joe Schmitt <joschmit@microsoft.com> 2.5.1-5
+* Mon Nov 22 2021 Thomas Crain <thcrain@microsoft.com> - 2.5.5-1
+- Upgrade to latest upstream version
+- Split out development files into devel subpackage
```
Comment (PawelWMS) [inline SPECS/libassuan/libassuan.spec:63]: Thanks for doing that!

File: SPECS/libksba/libksba.spec [hunk 3]
```diff
@@ -1,51 +1,67 @@
...
-License:	GPLv3+
-URL:		https://www.gnupg.org/(fr)/download/index.html#libksba
-Group:		Security/Libraries.
-Source0:        https://www.gnupg.org/ftp/gcrypt/%{name}/%{name}-%{version}.tar.bz2
+Summary:        Library for accessing X.509 and CMS data structure.
+Name:           libksba
+Version:        1.6.0
+Release:        1%{?dist}
+# See AUTHORS file for licensing details
+License:	    (LGPLv3+ or GPLv2+) and GPLv3+
```
Comment (PawelWMS) [inline SPECS/libksba/libksba.spec:6]: Nit: let's lint this file as well and remove tabs.

**Review by PawelWMS — APPROVED** [review]

File: SPECS/libksba/libksba.spec (hunk 3 above)
Comment (thcrain-msft) [reply SPECS/libksba/libksba.spec]: Thanks for catching those- fixed!

**Pull request merged**